src/agent.rs

```diff
@@ -554,7 +684,7 @@ pub trait Agent: Sized + 'static {
     fn connected(&mut self, _id: HandlerId) {}
 
     /// This method called on every incoming message.
-    fn handle(&mut self, msg: Self::Input, id: HandlerId);
+    fn handle(&mut self, msg: Self::Input, id: Option<HandlerId>);
```

Comment:
I'm torn on this. I like that it indicates that there is no one listening for responses on the other end but I don't like that Private and Job agents will always have a HandlerId but will still need to unwrap the option. Thoughts on using a reserved constant like usize::MAX to indicate that this was a dispatcher?

Reply:
That's a fair point with Private and Job agents. I do have my reservations about `usize::MAX` as an Id because it makes it easier to do the lazy thing and just reply using the Id instead of checking first if its a dispatcher. It wouldn't cause any bad behavior of the app, it just isn't obvious to developers that the messages they are sending won't find a target.

I think if `HandlerId` got an `is_dispatcher()` method, that would go some of the way towards making it easier for users to check if they are dealing with a bridge or a dispatcher.

Additionally, since HandlerId is opaque, could it itself use `Option<usize>` to encode  that it is a dispatcher, and let Yew handle the `None` case when calling send?